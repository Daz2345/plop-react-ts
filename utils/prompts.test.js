const {
  getNamePrompt,
  getTypePrompt,
  getUseReactIntlPrompt,
} = require('./prompts.js');
const plopConfiguration = require('./configuration.js');

describe('[PROMPTS]', () => {
  describe('getNamePrompt', () => {
    it('should return Name prompt object with default property set to "Button"', () => {
      expect(getNamePrompt(plopConfiguration, 'component').default).toEqual(
        plopConfiguration.defaultComponentName,
      );
    });
    it('should return Name prompt object with default property set to "Button"', () => {
      expect(getNamePrompt(plopConfiguration, 'container').default).toEqual(
        plopConfiguration.defaultContainerName,
      );
    });
  });
  describe('getTypePrompt', () => {
    it('should return Type prompt object with default property set to "defaultComponentType" and with an array of choices', () => {
      expect(
        getTypePrompt(plopConfiguration, ['test1', 'test2']),
      ).toMatchSnapshot();
      expect(
        getTypePrompt(plopConfiguration, ['test1', 'test2']).choices(),
      ).toEqual(['test1', 'test2']);
    });
  });
  describe('getUseReactIntlPrompt', () => {
    it('should return UsereactIntl prompt object with default property is equal to "plopConfig.useReactIntlByDefault"', () => {
      expect(getUseReactIntlPrompt(plopConfiguration)).toMatchSnapshot();
    });
  });
});
