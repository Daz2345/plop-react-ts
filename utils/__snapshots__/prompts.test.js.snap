// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[PROMPTS] getTypePrompt should return Type prompt object with default property set to "defaultComponentType" and with an array of choices 1`] = `
Object {
  "choices": [Function],
  "default": "PureComponent Class",
  "message": "Select the type of component",
  "name": "type",
  "type": "list",
}
`;

exports[`[PROMPTS] getUseReactIntlPrompt should return UsereactIntl prompt object with default property is equal to "plopConfig.useReactIntlByDefault" 1`] = `
Object {
  "default": true,
  "message": "Do you want to use ReactIntl?",
  "name": "useReactIntl",
  "type": "confirm",
}
`;
